### PR TITLE
Characterization - first processes

### PR DESCRIPTION
**Pull request by DavidCnly**

### CODE REVIEW

**Comment by DavidCnly** [comment]
Sorry, I had made all the requested changes in a separate commit, but I can't find where that's gone/what has happened it?

File: pyproject.toml [hunk 1]
```diff
@@ -11,7 +11,12 @@ description = "SolarMonitor Active Regtion Tracking (SMART)"
 requires-python = ">=3.9"
 readme = { file = "README.rst", content-type = "text/x-rst" }
 license = { file = "licenses/LICENSE.rst", content-type = "text/plain" }
-dependencies = []
+dependencies = [
+  "numpy",
+  "sunpy",
```
Comment (samaloney) [inline pyproject.toml:16]:
```suggestion
  "sunpy[map]",
```

File: smart/tests/test_functions.py [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    cosine_correction,
+    get_cosine_correction,
+    map_threshold,
+    smooth_los_threshold,
+)
+
+
+@pytest.fixture
+def hmi_nrt():
+    return "https://solmon.dias.ie/data/2024/06/06/HMI/fits/hmi.m_720s_nrt.20240606_230000_TAI.3.magnetogram.fits"
```
Comment (samaloney) [inline smart/tests/test_functions.py:19]: Prob makes sense to get the data directly from JSOC e.g. http://jsoc.stanford.edu/data/hmi/fits/

File: smart/characterization/dB_dt.py [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+                "mean B": B_mean.to(u.G),
+                "std B": B_std.to(u.G),
+                "minimum B": B_min.to(u.G),
+                "maximum B": B_max.to(u.G),
+                "positive flux": flux_pos.to(u.Wb),
+                "negative flux": flux_neg.to(u.Wb),
+                "unsigned flux": flux_uns.to(u.Wb),
+                "flux imbalance": flux_imb,
+                "total area": total_area.to(u.m**2),
+            }
```
Comment (samaloney) [inline smart/characterization/dB_dt.py:190]: Not sure about some of these particularly the unsigned flux have a look the idl code here

https://github.com/TCDSolar/SMART/blob/0530e19ba2aa7bc553153a97379460cae041e609/property_modules/ar_magprop.pro#L95-L113

**Pull request merged**